enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,544 @@
+
+### Resolveset Bundle Format
+
+A "resolveset" is an OLM bundle format that expresses the set of bundles that need to exist on the cluster in order to satisfy consumer constraints. While this resolveset bundle format is still in the work-in-progress stage, we can currently define this format as a YAML stream of individual BundleDeployment manifests.
+
+### Bundle/BundleDeployment
+
+A "BundleDeployment" is an OLM API that's responsible for managing OLM's first-class Bundle API, which represents content that needs to be made available to other consumers on the cluster. These APIs are analogous to the Deployment/PodSpec resources seen throughout core Kubernetes.
+The BundleDeployment API allows users to specify a Bundle template, and that template contains the desired specification of a Bundle resource.
+The management of a Bundle and BundleDeployment resources is delegated to individual "provisioner" implementations.
+
+### Provisioner
+
+A "provisioner" is a controller that understands the Bundle and BundleDeployment APIs. Individual provisioner implementations use ingress-style "class" name identifiers to distinguish between the BundleDeployments and Bundles resources that it needs to manage. Each provisioner implementation requires the definition of a unique provisioner IDs.
+This provisioner pattern helps abstract the variations of bundle formats that are defined throughout the Kubernetes landscape. Future provisioner implementations may manage helm charts, crossplane packages, and OLM bundle formats.
+
+Both the Bundle and BundleDeployment APIs require a unique provisioner ID to be specified. This enables higher level systems (e.g. a PO manager) to generate BundleDeployments that referencing different unique provisioner IDs. This is useful for components that want more control over how to source, and apply, Bundle contents.
+
+## Proposal
+
+This first phase will introduce a new, cluster-scoped PlatformOperator CRD, and that resource will enable admins to install an OLM-based operator during day 0 operations. A PO manager (POM) component will also be introduced, which is responsible for watching and reconciling the PlatformOperator custom resources.
+
+> Note: Given this is an openshift-only API, this proposal recommends introducing the CRD into the openshift/api repository under the `platform.openshift.io/v1alpha1` API group.
+
+During phase 0, this resource will contain an empty spec. Future phase requirements may necessitate more granular admin controls over how upgrades are approved, and how PO packages are sourced. During phase 0, there may be a limitation on which catalog sources are supported.
+
+Before cluster creation, an admin would be responsible for creating, and editing, the cluster's Kubernetes manifests with their desired PlatformOperator(s) configuration, and a FeatureGate manifest that enables tech preview features. The installer would then bootstrap the cluster using those custom manifests.
```

Comment:
I think that is what https://github.com/openshift/api/pull/1229 is meant to represent?

Reply:
Yep, the workflow for phase 0 would be admins explicitly enabling tech preview features before cluster creation through a FeatureGate resource, and that o/api PR you linked will add platform operators as a valid tech preview feature. Maybe this section is missing an explicit example for how you'd achieve that?

Reply:
The last commit removed an explicit feature flag for enabling POs through the TechPreviewNoUpgrade feature set. During phase 0, admins can enable POs by enabling tech preview features for the cluster.